plugins/connection/nsenter.py

```diff
@@ -0,0 +1,245 @@
+# (c) 2021 Jeff Goldschrafe <jeff@holyhandgrenade.org>
+# Based on Ansible local connection plugin by:
+# (c) 2012 Michael DeHaan <michael.dehaan@gmail.com>
+# (c) 2015, 2017 Toshio Kuratomi <tkuratomi@ansible.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: nsenter
+    short_description: execute on host running controller container
+    version_added: 1.9.0
+    description:
+        - This connection plugin allows Ansible, running in a privileged container, to execute tasks on the container
+          host instead of in the container itself.
+        - This is useful for running Ansible in a pull model, while still keeping the Ansible control node
+          containerized.
+        - It relies on having privileged access to run nsenter in the host's PID namespace, allowing it to enter the
+          namespaces of the provided PID (default PID 1, or init/systemd).
+    author: Jeff Goldschrafe (@jgoldschrafe)
+    options:
+        nsenter_pid:
+            description:
+                - PID to attach with using nsenter.
+                - The default should be fine unless you're attaching as a non-root user.
+            type: int
+            default: 1
+            vars:
+                - name: ansible_nsenter_pid
+            env:
+                - name: ANSIBLE_NSENTER_PID
+            ini:
+                - section: nsenter_connection
+                  key: nsenter_pid
+        pipelining:
+            default: ANSIBLE_PIPELINING
+            description:
+                - Pipelining reduces the number of connection operations required to execute a module on the remote
+                  server, by executing many Ansible modules without actual file transfers.
+                - This can result in a very significant performance improvement when enabled.
+                - However this can conflict with privilege escalation (become).
+                  For example, when using sudo operations you must first disable 'requiretty' in the sudoers file for
+                  the target hosts, which is why this feature is disabled by default.
+            env:
+                - name: ANSIBLE_PIPELINING
+            ini:
+                - section: defaults
+                  key: pipelining
+            type: boolean
+            vars:
+                - name: ansible_pipelining
+    notes:
+        - The remote user is ignored; this plugin always runs as root.
+        - "This plugin requires the Ansible controller container to be launched in the following way:"
+        - (1) The container image contains the nsenter program;
```

Comment:
Unfortunately this won't work either (at least not long-term), since once you include notes from a doc fragment, Ansible will sort the notes alphabetically, and these will end up at the top of the list without context :)